src/frontend/app/shared/components/list/list-types/cf-spaces-service-instances/cf-spaces-service-instances-list-config.service.ts

```diff
@@ -1,25 +1,21 @@
+import { DatePipe } from '@angular/common';
 import { Injectable } from '@angular/core';
-import { CfSpacesServiceInstancesDataSource } from './cf-spaces-service-instances-data-source';
-import { ListViewTypes, IListAction, ListConfig, IListConfig } from '../../list.component.types';
-import { ListView } from '../../../../../store/actions/list.actions';
 import { Store } from '@ngrx/store';
-import { AppState } from '../../../../../store/app-state';
+
+import { IServiceInstance } from '../../../../../core/cf-api-svc.types';
 import { CloudFoundrySpaceService } from '../../../../../features/cloud-foundry/services/cloud-foundry-space.service';
-import { APIResource } from '../../../../../store/types/api.types';
-import { ITableColumn } from '../../list-table/table.types';
-import { CfServiceInstance } from '../../../../../store/types/service.types';
-import { TableCellServiceNameComponent } from './table-cell-service-name/table-cell-service-name.component';
-import { TableCellServicePlanComponent } from './table-cell-service-plan/table-cell-service-plan.component';
-import { TableCellServiceInstanceTagsComponent } from './table-cell-service-instance-tags/table-cell-service-instance-tags.component';
-import {
-  TableCellServiceInstanceAppsAttachedComponent
-} from './table-cell-service-instance-apps-attached/table-cell-service-instance-apps-attached.component';
-import { DeleteServiceInstance, DeleteServiceBinding } from '../../../../../store/actions/service-instances.actions';
+import { ListView } from '../../../../../store/actions/list.actions';
 import { RouterNav } from '../../../../../store/actions/router.actions';
-import { DatePipe } from '@angular/common';
+import { DeleteServiceBinding, DeleteServiceInstance } from '../../../../../store/actions/service-instances.actions';
+import { AppState } from '../../../../../store/app-state';
+import { APIResource } from '../../../../../store/types/api.types';
+import { IListAction, IListConfig, ListConfig, ListViewTypes } from '../../list.component.types';
+import { ServiceInstancesListConfigService } from '../service-instances/service-instances-list-config.service';
+import { CfSpacesServiceInstancesDataSource } from './cf-spaces-service-instances-data-source';
+
 @Injectable()
-export class CfSpacesServiceInstancesListConfigService extends ListConfig<APIResource<CfServiceInstance>>
-  implements IListConfig<APIResource<CfServiceInstance>>  {
+export class CfSpacesServiceInstancesListConfigService extends ListConfig<APIResource<IServiceInstance>>
```

Comment:
Both this and `ServiceInstancesListConfigService` look identical apart from their datasource. Common stuff should be split into a base class, these two should extend that and then provide their own data sources.